app/src/androidTest/java/com/github/sdp/mediato/DatabaseTests.java

```diff
@@ -0,0 +1,110 @@
+package com.github.sdp.mediato;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import android.provider.ContactsContract;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.github.sdp.mediato.data.Database;
+import com.github.sdp.mediato.model.Location;
+import com.github.sdp.mediato.model.User;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+@RunWith(AndroidJUnit4.class)
+/**
+ * This class contains all the tests for database interactions
+ * @TODO add the Cloud Storage tests for the profile pictures
+ */
+public class DatabaseTests {
+    User user1;
+    User user2;
+    User user3;
+
+    @Before
+    public void setUp(){
+       try {
+           Database.database.useEmulator("10.0.2.2", 9000);
+        }
+        catch(Exception e){}
+        //Create new sample users
+        user1 = new User.UserBuilder("uniqueId1")
+                .setUsername("user_test_1")
+                .setEmail("email_test_1")
+                .setRegisterDate("09/03/2023")
+                .setLocation(new Location(3.14, 3.14))
+                .build();
+        user2 = new User.UserBuilder("uniqueId2")
+                .setUsername("user_test_2")
+                .setEmail("email_test_2")
+                .setRegisterDate("09/03/2023")
+                .setLocation(new Location(3.14, 3.14))
+                .build();
+        user3 = new User.UserBuilder("uniqueId3")
+                .setUsername("user_test_3")
+                .setEmail("email_test_3")
+                .setRegisterDate("09/03/2023")
+                .setLocation(new Location(3.14, 3.14))
+                .build();
+    }
+
+    @After
+    //Cleans database values after each test
+    public void cleanDatabase(){
+        Database.database.getReference().setValue(null);
+    }
+
+    @Test
+    //Tests that the user is properly added and retrieved from the database
+    public void addsAndGetsUserProperly() throws InterruptedException {
+        Database.addUser(user1).orTimeout(5, TimeUnit.SECONDS).join();

```

Comment:
would improve maintanability to use "5" as a constant variable in the tests rather than hardcode it as a magic number

Reply:
done :)